### PR TITLE
feat(cuda): Adds a parameter in the CUDA host functions passing the gpu index that should be used.

### PR DESCRIPTION
**Pull request by pdroalves**
### Checklist 

(Use '[x]' to check the checkboxes, or submit the PR and then click the checkboxes)

* [ ] Tests for the changes have been added (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)
* [x] The PR description links to the related issue (to link an issue, use '#XXX'.)
* [x] The tests on AWS have been launched and are successful (comment with @slab-ci cpu_test and/or @slab-ci gpu_test to trigger the tests)
* [ ] The draft release description has been updated
* [ ] Check for breaking changes (including serialization changes) and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

**Comment by pdroalves** [comment]
@slab-ci cpu_test
@slab-ci gpu_test

**Comment by pdroalves** [comment]
@slab-ci cpu_test
@slab-ci gpu_test

**Comment by pdroalves** [comment]
@slab-ci gpu_test

**Comment by pdroalves** [comment]
@slab-ci cpu_test
@slab-ci gpu_test

**Comment by pdroalves** [comment]
@agnesLeroy I can't figure out why that test is failing. Any idea?

File: .github/workflows/gpu_benchmark.yml [hunk 1]
```diff
@@ -32,7 +32,17 @@ jobs:
   run-cuda-benchmarks:
     name: Execute concrete-cuda benchmarks in EC2
     runs-on: ${{ github.event.inputs.runner_name }}
-    if: ${{ !cancelled() }}
+    strategy:
```
Comment (agnesLeroy) [inline .github/workflows/gpu_benchmark.yml:35]: This file shouldn't be modified, it seems to be a mistake no?

**Review by agnesLeroy — COMMENTED** [review]
Hi @pdroalves! Lgtm, there's just a file that has been accidentally modified I think.

**Comment by agnesLeroy** [comment]
There's also an issue given that the tests are failing: maybe you forgot to add the gpu_index argument somewhere, or passed it in the wrong order?

File: .github/workflows/gpu_benchmark.yml (hunk 1 above)
Comment (pdroalves) [reply .github/workflows/gpu_benchmark.yml:35]: Weird, this file is exactly the same as the one in the main branch.

```
➜  concrete-core git:(main) ✗ sha256sum .github/workflows/gpu_benchmark.yml
ced2dffdfc93301e5363062819a1897c65d55620d6f2181956932b92661ddb0a  .github/workflows/gpu_benchmark.yml
➜  concrete-core git:(main) ✗ git checkout cuda/feat/gpuindex
Switched to branch 'cuda/feat/gpuindex'
➜  concrete-core git:(cuda/feat/gpuindex) ✗ sha256sum .github/workflows/gpu_benchmark.yml
ced2dffdfc93301e5363062819a1897c65d55620d6f2181956932b92661ddb0a  .github/workflows/gpu_benchmark.yml
```

Comment (pdroalves) [reply .github/workflows/gpu_benchmark.yml]: Nvm, my local main was not updated. 
I rebased the branch.

**Comment by pdroalves** [comment]
@slab-ci cpu_test
@slab-ci gpu_test

**Comment by pdroalves** [comment]
@slab-ci cpu_test
@slab-ci gpu_test

**Comment by pdroalves** [comment]
Now it works!

**Review by agnesLeroy — APPROVED** [review]

**Pull request merged**